File: examples-sdk/zig/404-agent-data-csv-analyst/main.zig
// 404-agent-data-csv-analyst — Zig SDK equivalent
// Compile: vil compile --from zig --input 404-agent-data-csv-analyst/main.zig --release

const vil = @import("vil");

pub fn main() void {
    var server = vil.Server.init("csv-analyst-agent", 3123);
    var csv_analyst_agent = vil.Service.init("csv-analyst-agent");
    csv_analyst_agent.endpoint("POST", "/csv-analyze", "csv_analyze_handler");
    server.service(&csv_analyst_agent);
    server.compile();
}

File: examples-sdk/go/404-agent-data-csv-analyst/main.go
// 404-agent-data-csv-analyst — Go SDK equivalent
// Compile: vil compile --from go --input 404-agent-data-csv-analyst/main.go --release
package main

import vil "github.com/OceanOS-id/vil-go"

func main() {
	s := vil.NewServer("csv-analyst-agent", 3123)

	csv_analyst_agent := vil.NewService("csv-analyst-agent")
	csv_analyst_agent.Endpoint("POST", "/csv-analyze", "csv_analyze_handler")
	s.Service(csv_analyst_agent)

	s.Compile()
}
